cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -159,11 +160,32 @@ private void publish(String topicConfig, MonitorConsumeRequest request,
     messagePublisher.publish(NamespaceId.SYSTEM.getNamespace(), cConf.get(topicConfig),
                              messages.stream().map(s -> s.getMessage().getBytes(StandardCharsets.UTF_8)).iterator());
 
-    topicsToRequest.put(topicConfig, new MonitorConsumeRequest(request.getTopic(), topicConfig,
-                                                               messages.get(messages.size() - 1).getMessageId(),
+    topicsToRequest.put(topicConfig, new MonitorConsumeRequest(messages.get(messages.size() - 1).getMessageId(),
                                                                limit));
   }
 
+  private void triggerRuntimeShutdown() throws Exception {
+    try {
+      restClient.execute(HttpRequest.builder(HttpMethod.POST, clientConfig.resolveURL("runtime/shutdown")).build());
+    } catch (ConnectException e) {
+      LOG.trace("Connection refused when attempting to connect to Runtime Http Server. " +
+                  "Assuming that it is not available.");
+    }
+  }
+
+  private void setIsRuntimeInactive(List<MonitorMessage> monitorMessages) {
+    for (MonitorMessage message : monitorMessages) {
```

Comment:
yea but since it only loops for programstatus and we  donot expect a lot of messages for program status topic, looping twice will not create big overhead.

Reply:
Right, I have handled that by modifying client to throw exception if `monitorResponses` is null